src/Controls/src/SourceGen/CodeBehindGenerator.cs

```diff
@@ -0,0 +1,333 @@
+				if (symbol == null)
+					continue;
+				foreach (var attr in symbol.GetAttributes())
+				{
+					if (!SymbolEqualityComparer.Default.Equals(attr.AttributeClass, xmlnsDefinitonAttribute))
+						continue;
+					var xmlnsDef = new XmlnsDefinitionAttribute(attr.ConstructorArguments[0].Value as string, attr.ConstructorArguments[1].Value as string);
+					if (attr.NamedArguments.Length == 1 && attr.NamedArguments[0].Key == nameof(XmlnsDefinitionAttribute.AssemblyName))
+						xmlnsDef.AssemblyName = attr.NamedArguments[0].Value.Value as string;
+					else
+						xmlnsDef.AssemblyName = symbol.Name;
+					cache.Add(xmlnsDef);
+				}
+			}
+
+			return cache;
+		}
+
+		bool TryGenerateXamlCodeBehind(AdditionalText file, GeneratorExecutionContext context, IList<XmlnsDefinitionAttribute> xmlnsDefinitionCache, out string? code)
+		{
+			code = null;
+			using (var reader = File.OpenText(file.Path))
+			{
+				if (!TryParseXaml(reader, context, xmlnsDefinitionCache, out var rootType, out var rootClrNamespace, out var generateDefaultCtor, out var addXamlCompilationAttribute, out var hideFromIntellisense, out var XamlResourceIdOnly, out var baseType, out var namedFields))
+					return false;
+
+				var sb = new StringBuilder();
+				var opt = context.AnalyzerConfigOptions.GetOptions(file);
+				if (   context.AnalyzerConfigOptions.GetOptions(file).TryGetValue("build_metadata.additionalfiles.ManifestResourceName", out string? manifestResourceName)
+					&& context.AnalyzerConfigOptions.GetOptions(file).TryGetValue("build_metadata.additionalfiles.TargetPath", out string? targetPath))
+					sb.AppendLine($"[assembly: global::Microsoft.Maui.Controls.Xaml.XamlResourceIdAttribute(\"{manifestResourceName}\", \"{targetPath.Replace('\\','/')}\", typeof({baseType}))]");
+				if (XamlResourceIdOnly)
+				{
+					code = sb.ToString();
+					return true;
+				}
+				if (rootType == null)
+					throw new Exception("Something went wrong");
+
+				sb.AppendLine($"namespace {rootClrNamespace}");
+				sb.AppendLine("{");
+
+				if (context.AnalyzerConfigOptions.GetOptions(file).TryGetValue("build_metadata.additionalfiles.ItemSpec", out string? itemSpec))
+					sb.AppendLine($"\t[global::Microsoft.Maui.Controls.Xaml.XamlFilePathAttribute(\"{itemSpec.Replace("\\", "\\\\")}\")]");
+				if (addXamlCompilationAttribute)
+					sb.AppendLine($"\t[global::Microsoft.Maui.Controls.Xaml.XamlCompilationAttribute(global::Microsoft.Maui.Controls.Xaml.XamlCompilationOptions.Compile)]");
+				if (hideFromIntellisense)
+					sb.AppendLine($"\t[global::System.ComponentModel.EditorBrowseableAttribute(global::System.ComponentModel.EditorBrowsableState.Never)]");
+
+				sb.AppendLine($"\tpublic partial class {rootType} : {baseType}");
+				sb.AppendLine("\t{");
+				
+				//optional default ctor
+				if (generateDefaultCtor)
+				{
+					sb.AppendLine($"\t\t[global::System.CodeDom.Compiler.GeneratedCodeAttribute(\"Microsoft.Maui.Controls.SourceGen\", \"1.0.0.0\")]");
```

Comment:
Should we apply the GeneratedCodeAttribute to the class itself too?

Reply:
we only add the attribute to the class is it's codebehind for a code-less ResourceDictionary